packages/java/endpoint/src/main/java/com/vaadin/hilla/route/RouteUnifyingIndexHtmlRequestListener.java

```diff
@@ -207,4 +253,37 @@ private Map<String, RouteParamType> getRouteParameters(
         });
         return routeParameters;
     }
+
+    boolean isClientMenuUsed(VaadinService vaadinService) {
+        try {
+            String fileRoutesJson = readFileRoutesJsonFile(
+                    ApplicationConfiguration.get(vaadinService.getContext()));
+            var json = mapper.readTree(fileRoutesJson);
+            // as the client registry has no information about layouts, the JSON
+            // is parsed again to search for a root node with children, which
+            // means that we have a root layout.
+            return json.has(0) && json.get(0).has("children")
+                    && json.get(0).get("children").isArray()
+                    && !json.get(0).get("children").isEmpty();
```

Comment:
@cromoteca I think we need to also check that the `json.get(0).get("route")` is empty, and also this last check is not needed here since a main layout should satisfy the following criteria:
1. It should have a `"route": ""`, and
2. It should have a `"children": []` or `"children": [...]`

WDYT?

Reply:
Now that I look again, for checking the existence of a main layout, I suggest not to read the `file-routes.json` again, as it is done once for when `ClientRouteRegistry` registering the routes. Based on what we already have in `registeredRoutes` (in ClientRouteRegistry) we can determine whether a main layout with empty string route ("") and a non-null children property exists or not.

Reply:
Using the `registeredRoutes` map would be indeed more practical, considering that the main layout might not be the first item.